sceptre/synapsedev/templates/testCMK.json

```diff
@@ -0,0 +1,104 @@
+                "EnableKeyRotation": false,
+                "KeyPolicy": {
+                    "Version": "2012-10-17",
+                    "Id": "key-default-1",
+                    "Statement": [
+                        {
+                            "Sid": "Allow administration of the key",
+                            "Effect": "Deny",
+                            "Principal": {
+                                "AWS": "*"
+                            },
+                            "Action": [
+                                "kms:*"
+                            ],
+                            "Resource": "*",
+                            "Condition": {
+                                "StringNotLike": {
+                                    "aws:PrincipalArn": [
+                                        {
+                                            "Fn::Join": [
+                                                "",
+                                                [
+                                                    "arn:aws:iam::",
+                                                    {
+                                                        "Ref": "AWS::AccountId"
+                                                    },
+                                                    ":root"
+                                                ]
+                                            ]
+                                        },
+                                        "arn:aws:iam::449435941126:role/*"
+                                    ]
+                                }
+                            }
+                        },
+                        {
+                            "Sid": "Allow root administration of the key",
+                            "Effect": "Allow",
+                            "Principal": {
+                                "AWS": [
+                                    {
+                                        "Fn::ImportValue": "us-east-1-accounts-AWSIAMAdminRoleArn"
+                                    },
+                                    {
+                                        "Fn::Join": [
+                                            "",
+                                            [
+                                                "arn:aws:iam::",
+                                                {
+                                                    "Ref": "AWS::AccountId"
+                                                },
+                                                ":root"
+                                            ]
+                                        ]
+                                    },
+                                    "arn:aws:sts::449435941126:assumed-role/AWSReservedSSO_Administrator_693a85eb20cd5043/x.schildwachter@sagebase.org",
+                                    "arn:aws:iam::449435941126:role/sagebase-github-oidc-sage-ProviderRoleorganization-1K1MGUYPL5JUO"
```

Comment:
you just denied all except for root in the previous policy.  deny overrides allow therefore I don't think this will allow these principals.

Reply:
At line 45, any role should be excepted as well. It worked for SSO, I want to see if it's gonna work for oidc.